server/src/middleware/rpc/mod.rs

```diff
@@ -26,24 +26,30 @@
 
 //! Various middleware implementations for JSON-RPC specific purposes.
 
-mod layers;
+pub mod layer;
+pub use layer::*;
 
-pub use layers::*;
+use futures_util::Future;
+use jsonrpsee_core::server::MethodResponse;
+use jsonrpsee_types::Request;
+use layer::either::Either;
 
 use tower::layer::util::{Identity, Stack};
 use tower::layer::LayerFn;
 
-use jsonrpsee_core::server::MethodResponse;
-use jsonrpsee_types::Request;
-
 /// Similar to `tower::Service` but specific for jsonrpsee and
 /// doesn't requires `&mut self` for performance reasons.
-#[async_trait::async_trait]
 pub trait RpcServiceT<'a> {
+	/// The future response value.
+	//
+	// type Future<'cx>: Future<Output = MethodResponse> where Self: 'cx, 'a: 'cx;
+	// would be better but ran into https://github.com/rust-lang/rust/issues/100013
+	type Future: Future<Output = MethodResponse> + Send;
```

Comment:
TODO: remove send bound

Reply:
I changed my mind it's a mess to change that and every future in jsonrpsee requires Send